tests/testthat/test-generate_pt_nowcast_mat_list.R

```diff
@@ -144,6 +144,70 @@ test_that("generate_pt_nowcast_mat_list identical-sized matrices work", {
   expect_identical(sapply(result, nrow), c(6L, 6L))
 })
 
+test_that("Function handles a single triangle with 0s for first column appropriately", { # nolint
+
+  triangle3 <- matrix(
+    c(
+      0, 40, 20, 5,
+      0, 50, 10, 10,
+      0, 40, 31, NA,
+      0, 45, NA, NA,
+      0, NA, NA, NA
+    ),
+    nrow = 5,
+    byrow = TRUE
+  )
+
+  retro_rts_list <- list(test_triangle_1, test_triangle_2, triangle3)
+
+  result <- expect_message(generate_pt_nowcast_mat_list(retro_rts_list))
+  expect_null(result[[3]])
+})
+
+test_that("Function errors if only contains triangles with first column 0", { # nolint
```

Comment:
function name here and elsewher.

Reply:
I changed all the testthat strings to use the function name in the files edited here. I think we should streamline in a separate issue for remainder.